ttnn/cpp/ttnn/tensor/types.cpp

```diff
@@ -293,3 +293,28 @@ MemoryConfig load_memory_config(const std::string& file_name) {
 }  // namespace tt_metal
 
 }  // namespace tt
+
+namespace ttnn {
+namespace types {
+
+uint32_t &Shape::operator[](const std::int64_t index) {
+    auto normalized_index = this->get_normalized_index(index);
+    return this->value[normalized_index];
+}
+
+const uint32_t Shape::operator[](std::int64_t index) const { return this->value.without_padding()[index]; }
+
+const uint32_t* Shape::begin() const {return this->value.begin(); }
+const uint32_t* Shape::end() const {return this-> value.end(); }
```

Comment:
this is inconsistent with operator[] behavior

Reply:
I can change to `this->value.without_padding().begin()`